### ISSUE TITLE
Gtk3: need testing of rich copy-paste *to* zim

### ISSUE THREAD

**Issue by jaap-karssenberg** [jaap-karssenberg/zim-desktop-wiki#332]
As reported in bug #326 rich copy paste *from* zim is broken. Rich copy paste *to* zim should still work, but cannot be tested automatically.

Need (manual) testing for pasting into a page
- images
- links
- text

Automated tests using other tools to set the clipboard are welcome as well (and may provide clues to solving #326 as well ...)

**Comment by muguu** [comment]
Test for pasting an image into a page: FAIL
`DEBUG: Action: paste
DEBUG: Targets available for paste: (True, targets=[Gdk.Atom.intern("TIMESTAMP", False), Gdk.Atom.intern("TARGETS", False), Gdk.Atom.intern("MULTIPLE", False), Gdk.Atom.intern("SAVE_TARGETS", False), Gdk.Atom.intern("text/html", False), Gdk.Atom.intern("text/_moz_htmlinfo", False), Gdk.Atom.intern("text/_moz_htmlcontext", False), Gdk.Atom.intern("image/png", False), Gdk.Atom.intern("image/bmp", False), Gdk.Atom.intern("image/x-bmp", False), Gdk.Atom.intern("image/x-MS-bmp", False), Gdk.Atom.intern("image/x-icon", False), Gdk.Atom.intern("image/x-ico", False), Gdk.Atom.intern("image/x-win-bitmap", False), Gdk.Atom.intern("image/vnd.microsoft.icon", False), Gdk.Atom.intern("application/ico", False), Gdk.Atom.intern("image/ico", False), Gdk.Atom.intern("image/icon", False), Gdk.Atom.intern("text/ico", False), Gdk.Atom.intern("image/jpeg", False), Gdk.Atom.intern("image/tiff", False)]), we want parsetree
WARNING: Could not paste - no compatible data types on clipboard`


(Same test with master was successful): 
`DEBUG: Requesting data for image/png
DEBUG: Saving image from clipboard to /home/murat/Notebooks/Notes/Home/pasted_image.png
DEBUG: Pasting link: file:///home/murat/Notebooks/Notes/Home/pasted_image.png (type: file, isimage: True)`


Complete Debug:
```
INFO: This is zim 0.68-gtk3
DEBUG: Python version is sys.version_info(major=3, minor=6, micro=3, releaselevel='final', serial=0)
DEBUG: Platform is posix
DEBUG: No bzr version-info found
DEBUG: Running from a source dir: /home/murat/zim-desktop-wiki-next
DEBUG: Set XDG_DATA_HOME to /home/murat/.local/share
DEBUG: Set XDG_DATA_DIRS to [<Dir: /usr/share/ubuntu>, <Dir: /usr/share/ubuntu>, <Dir: /usr/local/share>, <Dir: /usr/share>, <Dir: /var/lib/snapd/desktop>]
DEBUG: Set XDG_CONFIG_HOME to /home/murat/.config
DEBUG: Set XDG_CONFIG_DIRS to [<Dir: /etc/xdg/xdg-ubuntu>, <Dir: /etc/xdg/xdg-ubuntu>, <Dir: /etc/xdg>]
DEBUG: Set XDG_CACHE_HOME to /home/murat/.cache
DEBUG: Connecting to /tmp/zim-murat/primary-0.68-gtk3-c96ad963
DEBUG: Got error in dispatch: No such file or directory
DEBUG: Starting primary process
DEBUG: Start listening on: /tmp/zim-murat/primary-0.68-gtk3-c96ad963
DEBUG: Registered object line
DEBUG: Loading config from: <zim.notebook.info.VirtualFile object at 0x7f73a971f668>
DEBUG: Loading config from: /home/murat/Notebooks/Notes/notebook.zim
DEBUG: Loading config from: /home/murat/Notebooks/Notes/notebook.zim
DEBUG: Loading config from: <ConfigFile: /home/murat/.config/zim/preferences.conf>
DEBUG: Loading plugin: calendar
DEBUG: Loading plugin: insertsymbol
DEBUG: Loading plugin: pageindex
DEBUG: Loading plugin: pathbar
DEBUG: Loading plugin: printtobrowser
DEBUG: Loading plugin: tasklist
DEBUG: Loading config from: /home/murat/.cache/zim/notebook-home_murat_Notebooks_Notes/state.conf
DEBUG: Loading config from: <ConfigFile: /home/murat/.config/zim/style.conf>
DEBUG: Autosave interval: 15 - use threads: True
INFO: Open page: Journal:2018:02:27:Home (Journal:2018:02:27:Home)
DEBUG: Action: toggle_panes(False)
DEBUG: Action: toggle_panes(True)
DEBUG: Accelmap: /home/murat/.config/zim/accelmap
DEBUG: Add window: MainWindow
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck found out-of-date
DEBUG: Index folder: /home/murat/Notebooks/Notes/Journal/2018/02/27
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
DEBUG: Store page in background: Journal:2018:02:27:Home
DEBUG: Index file: /home/murat/Notebooks/Notes/Journal/2018/02/27/Home.txt
DEBUG: Action: paste
DEBUG: Targets available for paste: (True, targets=[Gdk.Atom.intern("TIMESTAMP", False), Gdk.Atom.intern("TARGETS", False), Gdk.Atom.intern("MULTIPLE", False), Gdk.Atom.intern("SAVE_TARGETS", False), Gdk.Atom.intern("text/html", False), Gdk.Atom.intern("text/_moz_htmlinfo", False), Gdk.Atom.intern("text/_moz_htmlcontext", False), Gdk.Atom.intern("image/png", False), Gdk.Atom.intern("image/bmp", False), Gdk.Atom.intern("image/x-bmp", False), Gdk.Atom.intern("image/x-MS-bmp", False), Gdk.Atom.intern("image/x-icon", False), Gdk.Atom.intern("image/x-ico", False), Gdk.Atom.intern("image/x-win-bitmap", False), Gdk.Atom.intern("image/vnd.microsoft.icon", False), Gdk.Atom.intern("application/ico", False), Gdk.Atom.intern("image/ico", False), Gdk.Atom.intern("image/icon", False), Gdk.Atom.intern("text/ico", False), Gdk.Atom.intern("image/jpeg", False), Gdk.Atom.intern("image/tiff", False)]), we want parsetree
WARNING: Could not paste - no compatible data types on clipboard
```

**Comment by muguu** [comment]
Test for pasting **a text** into a page: FAIL

```
DEBUG: Action: paste
DEBUG: Targets available for paste: (True, targets=[Gdk.Atom.intern("TIMESTAMP", False), Gdk.Atom.intern("TARGETS", False), Gdk.Atom.intern("MULTIPLE", False), Gdk.Atom.intern("SAVE_TARGETS", False), Gdk.Atom.intern("GTK_TEXT_BUFFER_CONTENTS", False), Gdk.Atom.intern("application/x-gtk-text-buffer-rich-text", False), Gdk.Atom.intern("UTF8_STRING", False), Gdk.Atom.intern("COMPOUND_TEXT", False), Gdk.Atom.intern("TEXT", False), Gdk.Atom.intern("STRING", False), Gdk.Atom.intern("text/plain;charset=utf-8", False), Gdk.Atom.intern("text/plain", False)]), we want parsetree
WARNING: Could not paste - no compatible data types on clipboard
```

**Comment by genodeftest** [comment]
These should be fixed in #392.

**Comment by jaap-karssenberg** [comment]
Should be fixed with e42ebc605620567cb7ecc16a8551b038d1c36aae
